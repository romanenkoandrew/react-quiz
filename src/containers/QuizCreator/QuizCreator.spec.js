import React from "react";
import { shallow } from "enzyme";
import { QuizCreator } from "./QuizCreator";
import QuizCreatorConnect from "./QuizCreator";
import configureStore from "redux-mock-store";

const mockStore = configureStore();

describe("QuizCreator component", () => {
  let createQuizQuestion, finishCreateQuiz, component, instance;
  beforeEach(() => {
    createQuizQuestion = jest.fn();
    finishCreateQuiz = jest.fn();
    component = shallow(
      <QuizCreator
        quiz={[{ answers: "answers", question: "question" }]}
        createQuizQuestion={createQuizQuestion}
        finishCreateQuiz={finishCreateQuiz}
      />
    );
    instance = component.instance();
  });
  it("should render QuizCreator with props", () => {
    expect(component).toMatchSnapshot();
  });
  it("should call method onSubmitHandler in QuizCreator", () => {
    const e = { preventDefault: jest.fn() };
    instance.onSubmitHandler(e);
    expect(e.preventDefault).toBeCalledTimes(1);
  });
  it("should call method addQuestionHandler in QuizCreator", () => {
    const e = { preventDefault: jest.fn() };
    instance.addQuestionHandler(e);
    expect(e.preventDefault).toBeCalledTimes(1);
  });
  it("should call method createQuizHandler in QuizCreator", () => {
    const e = { preventDefault: jest.fn() };
    instance.createQuizHandler(e);
    expect(e.preventDefault).toBeCalledTimes(1);
  });
  it("should call method selectChangeHandler  in QuizCreator", () => {
    const e = { target: { value: 1 } };
    instance.selectChangeHandler(e);
    expect(instance.state.rightAnswerId).toBe(1);
  });
  it("should call method onChangeHandler in QuizCreator", () => {
    instance.setState({ formControls: { controlName: {} } });
    instance.onChangeHandler("value", "controlName");
    expect(instance.state.formControls.controlName.touched).toBe(true);
    expect(instance.state.formControls.controlName.value).toBe("value");
    expect(instance.state.formControls.controlName.valid).toBe(true);
    expect(instance.state.isFormValid).toBe(true);
  });
});

describe("QuizCreatorConnect component", () => {
  it("should props from store in QuizCreatorConnect", () => {
    const store = mockStore({ createQuiz: { quiz: {} } });
    const component = shallow(<QuizCreatorConnect store={store} />);
    expect(component.children().props().quiz).toBeTruthy();
  });
});
