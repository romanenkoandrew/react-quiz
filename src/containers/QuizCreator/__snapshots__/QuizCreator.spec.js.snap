// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizCreator component should render QuizCreator with props 1`] = `
<div
  className="QuizCreator"
>
  <div>
    <h1>
      Создание теста
    </h1>
    <form
      onSubmit={[Function]}
    >
      <Auxiliary
        key="question0"
      >
        <Input
          errorMessage="Вопрос не может быть пустым"
          label="Введите вопрос"
          onChange={[Function]}
          shouldValidate={true}
          touched={false}
          valid={false}
          value=""
        />
        <hr />
      </Auxiliary>
      <Auxiliary
        key="option11"
      >
        <Input
          errorMessage="Значение не может быть пустым"
          label="Вариант 1"
          onChange={[Function]}
          shouldValidate={true}
          touched={false}
          valid={false}
          value=""
        />
      </Auxiliary>
      <Auxiliary
        key="option22"
      >
        <Input
          errorMessage="Значение не может быть пустым"
          label="Вариант 2"
          onChange={[Function]}
          shouldValidate={true}
          touched={false}
          valid={false}
          value=""
        />
      </Auxiliary>
      <Auxiliary
        key="option33"
      >
        <Input
          errorMessage="Значение не может быть пустым"
          label="Вариант 3"
          onChange={[Function]}
          shouldValidate={true}
          touched={false}
          valid={false}
          value=""
        />
      </Auxiliary>
      <Auxiliary
        key="option44"
      >
        <Input
          errorMessage="Значение не может быть пустым"
          label="Вариант 4"
          onChange={[Function]}
          shouldValidate={true}
          touched={false}
          valid={false}
          value=""
        />
      </Auxiliary>
      <Select
        label="Выберите правильный ответ"
        onChange={[Function]}
        options={
          Array [
            Object {
              "text": "1",
              "value": 1,
            },
            Object {
              "text": "2",
              "value": 2,
            },
            Object {
              "text": "3",
              "value": 3,
            },
            Object {
              "text": "4",
              "value": 4,
            },
          ]
        }
        value={1}
      />
      <Button
        disabled={true}
        onClick={[Function]}
        type="primary"
      >
        Добавить вопрос
      </Button>
      <Button
        disabled={false}
        onClick={[Function]}
        type="success"
      >
        Создать тест
      </Button>
    </form>
  </div>
</div>
`;
